---
title: "Rockefeller advanced gene mapping course"
author: "Matthew Moll"
output: 
  html_document:
    toc: true
    toc_float: true
---


date: `r Sys.Date()`



## R setup
```{r setup}


knitr::opts_chunk$set(echo = TRUE, fig.width = 6, fig.height = 6)

library(tidyverse)
library(data.table)
library(knitr)
library(readxl)
library(epiR)
library(e1071)


```

## Load data
```{r load data}

rm(list = ls())


```

# Day 1: 

##GWAS, case/control and family-based studies

### lecture 1: Heather cordell

<BR>

**General concepts**
<BR>

* GWAS case and controls - easier than family based studies. However, family may be more informative

* May use population based controls for a rare disease, instead of matched, verified controls

* Use X2 test to look for association.  

* Can also do odds ratios: OR (2|2 : 1|1), OR (1|2: 1|1)

* Can also calculate genotype relative risk

* These tests can be done by alleles instead of genotypes as well

* Can also do logistic regression. Will compare to reference genotype so that 1/1 has OR of 1. 
  
    1/1 = 0 
    1/2 = 1
    2/2 = 2

* To model a dominant or rececessive model:

    Dominant- code 1/1 as 0, 1/2 and 2/2 as 1  
  
    Recessive - code 1/1 as 0, 1/2 as 0, and 2/2 as 1
    
* manhattan plot - want to show -log(p value) so that higher values are more important (lower p-value)

* p-value threshold usually around 5 X 10^-8

* need to have replication in external cohort

* QC

    + stringent QC checks needed
    
    + Discard if: 
        - low call rates
        - excess heterozygosity
        - X chr markers useful for gecking gender (males look homozygous for X)
        - Check relationships and ethnicity
        
    + Dischard if SNPs deemed unreliable
        - basis of call rates, mendelian misinheritances, HW disequilibrium
        - low minor allele frequency
        
    + Make plot of call rates vs heterozygosity to find outliers
    
    + Ethnicity tests - multidimensional scaling (210 HapMap individuals) to find individuals with non-Caucasion ancestry
        -helps to get homogeneous population if exclude outlier samples
    
    + Full vs half sibs
        - mean vs. standard error plots

<BR>

### Lecture 2: Leal: QC continued

<BR>

* Genotyping

    + check frequencies for all variant types in cases and controls, and batches to make sure no artifact
    + be very suspicious of one highly associated SNP without other SNPs nearby in LD being associated
    + remove individuals with > 3% of genotypes.  Then remove markers with > 5% missing genotypes.  
        - then amongst markers with maf < 5%, remove if >1% of genotypes missing
    + check for double recombination based on genotype distances - use Merlin for family data
    + check for non-mendelian errors of segregation - use Pedcheck for family data

<BR>

* Data cleaning

     + double check sex - true males will have homozygosity of X chr
         - if sex is mislabelled, best to assume this was a mislabelled sample, and remove this individual altogether
     + Look for cross contamination = cluster together in pca
     + check for duplicate and related individuals and remove. 
         - cryptic duplicates - DNA smaple aliquoted more than once, or individual enrolled twice in study
         - related individuals recruited to study
     + PLINK or KING can be used to look for cryptic relatives
     + PCA/MDS analysis to look for related individuals and outliers, as well as how well clusters separate
     
    
    
    

